tests/test_ec2/test_elastic_network_interfaces.py

```diff
@@ -401,6 +401,43 @@ def test_elastic_network_interfaces_get_by_description():
     enis.should.have.length_of(0)
 
 
+@mock_ec2_deprecated
+@mock_ec2
+def test_elastic_network_interfaces_get_by_attachment_instance_id():
+    conn = boto.connect_vpc("the_key", "the_secret")
```

Comment:
Creating the VPC can be done with the boto3-client as well.

`@mock_ec2_deprecated' makes it possible to mock boto, but were trying to phase out support for boto at some point. So new tests shouldn't have any `@.._deprecated` decorators.

Reply:
Do you have an example where we're using the new style and attaching an ENI? The only other ENI attach example I could find was using the deprecated style.

Reply:
It is be very similar using boto3. Haven't tested it out, but this should work:
```
eni1 = ec2_client.create_network_interface(SubnetId=..., Groups=[..])
ec2_client.attach_network_interface(DeviceIndex=1, InstanceId=..., NetworkInterfaceId=...)
```